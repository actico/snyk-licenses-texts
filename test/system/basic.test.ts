import { exec } from 'child_process';
import { sep } from 'path';
const main = './dist/index.js'.replace(/\//g, sep);

describe('`snyk-licenses-report help <...>`', () => {
  const OLD_ENV = process.env;
  process.env.SNYK_TOKEN = process.env.SNYK_TEST_TOKEN;
  afterAll(async () => {
    process.env = { ...OLD_ENV };
  });
  it('Shows help text as expected', async (done) => {
    return exec(`node ${main} help`, (err, stdout) => {
      if (err) {
        throw err;
      }
      expect(err).toBeNull();
      expect(stdout.trim()).toMatchSnapshot();
      done();
    });
  });
});
