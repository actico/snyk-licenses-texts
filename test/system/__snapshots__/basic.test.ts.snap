// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`snyk-licenses-report help <...>\` Shows help text as expected 1`] = `
"index.js <command>

Commands:
  index.js generate  Generate org licenses & dependencies report in HTML format
                                                                    [aliases: g]
  index.js json      Generate org licenses & dependencies data in JSON format
                                                                    [aliases: j]

Options:
  --version  Show version number                                       [boolean]
  --help     Show help                                                 [boolean]"
`;
